File: lib/plugins/lmgtfy.js
module.exports = function(bot, config) {
  
  bot.hear(/lmgtfy (.*)$/i, function(msg, reply) {
    var link = "http://lmgtfy.com/?q=" + encodeURIComponent(msg.match[1]);
    reply(link);
  });

  bot.help("lmgtfy", function(msg, reply) {
    reply("Syntax: lmgtfy <something>");
  });

};

File: plugins/lmgtfy.py
import urllib.parse


class Plugin:
    def __call__(self, bot):
        bot.on_respond(r"lmgtfy (.*)$", self.on_respond)
        bot.on_help("lmgtfy", self.on_help)

    def on_respond(self, bot, msg, reply):
        url = "http://lmgtfy.com/?q={0}".format(urllib.parse.quote(msg["match"][0]))
        reply(url)

    def on_help(self, bot, msg, reply):
        reply("Syntax: lmgtfy <query>")
